inc/t_cose/t_cose_common.h

```diff
@@ -220,6 +220,39 @@ extern "C" {
 /* Definition of struct t_cose_key is moved to t_cose_key.h */
 
 
+/**
+ * The size of the output of SHA-256.
+ *
+ * (It is safe to define these independently here as they are
+ * well-known and fixed. There is no need to reference
+ * platform-specific headers and incur messy dependence.)
+ */
+#define T_COSE_CRYPTO_SHA256_SIZE 32
+
+/**
+ * The size of the output of SHA-384 in bytes.
+ */
+#define T_COSE_CRYPTO_SHA384_SIZE 48
+
+/**
+ * The size of the output of SHA-512 in bytes.
+ */
+#define T_COSE_CRYPTO_SHA512_SIZE 64
+
+/**
+ * The maximum needed to hold a hash. It is smaller and less stack is needed
+ * if the larger hashes are disabled.
+ */
+#if !defined(T_COSE_DISABLE_ES512) || !defined(T_COSE_DISABLE_PS512)
```

Comment:
This calculation of the MAX_HASH_SIZE should not be here in common.h .   I didn't realize it until now, but it's not really global anymore. Each signer/verifier should have their own max based on what algorithms they support. 

I think have T_COSE_CRYPTO_SHA256_SIZE in the public common location is good and helpful for signer/verifier implementors.